fuzzing/config/config.go

```diff
@@ -112,11 +109,53 @@ type FuzzingConfig struct {
 	TestChainConfig config.TestChainConfig `json:"chainConfig"`
 }
 
-// fuzzingConfigMarshaling is a structure that overrides field types during JSON marshaling. It allows FuzzingConfig to
-// have its custom marshaling methods auto-generated and will handle type conversions for serialization purposes.
-// For example, this enables serialization of big.Int but specifying a different field type to control serialization.
-type fuzzingConfigMarshaling struct {
-	TargetContractsBalances []*hexutil.Big
+type ContractBalance struct {
+	big.Int
+}
+
+// UnmarshalJSON parses JSON data into big.Int from empty strings, hex ("0x"),
+// scientific notation (e/E), and base-10 formats
+
+func (cb *ContractBalance) UnmarshalJSON(data []byte) error {
+	s := strings.Trim(string(data), "\"")
+
+	// Empty string handling
+	if s == "" {
+		cb.Int.SetInt64(0)
+		return nil
+	}
+
+	// Hex notation handling
+	if strings.HasPrefix(strings.ToLower(s), "0x") {
+		if _, ok := cb.Int.SetString(s[2:], 16); !ok {
+			return fmt.Errorf("invalid hex string provided while unmarshaling contract balance: %s", s)
+		}
+		return nil
+	}
+
+	// Scientific notation handling
+	if strings.ContainsAny(s, "eE") {
+		f, err := strconv.ParseFloat(s, 64)
+		if err == nil {
+			plainStr := strconv.FormatFloat(f, 'f', 0, 64)
+			if _, ok := cb.Int.SetString(plainStr, 10); !ok {
+				return fmt.Errorf("invalid format for contract balance (scientific notation) while unmarshaling contract balance: %s", s)
+			}
+			return nil
+		} else {
+			return fmt.Errorf("error parsing scientific notation while unmarshaling contract balance: %w", err)
+		}
```

Comment:
This is another nitpick but generally you want to handle the `err != nil` case first instead of the success case. This reduces the nesting structure of this logic. Since if `err != nil`, then you will return an error and then you don't really need an else branch. This sort of "guard clause" structure reduces the amount of indentation and makes the code cleaner.